### PR TITLE
fix: revised zod schemas

### LINKED ISSUE #312: Revise zod schemas
## Description

The current [schemas](https://github.com/Tascurator/tascurator-frontend/blob/develop/src/constants/schema.ts) are not perfect in terms of validations. For example, some properties do not have `trim()`, which means they can contain whitespace before and after the value. This could potentially lead to bugs for values that must be exactly the same.

## Criteria

- [x] Add `trim()` to all string values to clean up any leading or trailing whitespace
- [x] Add `uuid()` to all string UUIDs

### PR DESCRIPTION
**Pull request by yuki-92c**
## Overview

- I revised the zod schemas by adding trim and uuid to missing lines

## Changes

- add `.trim()` to string attributes
- add `.uuid()` to uuid attributes

## Assignee Checklist:

- [x] The base branch is correct (no accidental merges)
- [x] The branch name follows our branch naming rules
- [x] The PR title follows our PR title rules
- [x] My code follows our coding style
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors in the console

## Reviewer Checklist:

- [ ] Code readability and simplicity
- [ ] Follows best practices and coding standards
- [ ] Understandable and maintainable code

### CODE REVIEW

File: src/constants/schema.ts [hunk 1]
```diff
@@ -217,7 +227,7 @@ export const signupSchema = z.object({
 export type TSignupSchema = z.infer<typeof signupSchema>;
 
 export const forgotPasswordSchema = z.object({
-  email: z.string().email(ERROR_MESSAGES.EMAIL_INVALID),
+  email: z.string().email(ERROR_MESSAGES.EMAIL_INVALID).trim(),
```
Comment (akiomatic) [inline src/constants/schema.ts]: The`.trim` should be before the `.email`.

File: src/constants/schema.ts [hunk 2]
```diff
@@ -194,18 +202,20 @@ export const rotationCycleUpdateSchema = z.object({
...
     .string()
+    .trim()
     .min(PASSWORD_MIN_LENGTH, minLength('Password', PASSWORD_MIN_LENGTH)),
 });
 
 export type TLoginSchema = z.infer<typeof loginSchema>;
 
 export const signupSchema = z.object({
-  email: z.string().email(ERROR_MESSAGES.EMAIL_INVALID),
+  email: z.string().email(ERROR_MESSAGES.EMAIL_INVALID).trim(),
```
Comment (akiomatic) [inline src/constants/schema.ts]: The`.trim` should be before the `.email`.

File: src/constants/schema.ts [hunk 3]
```diff
@@ -194,18 +202,20 @@ export const rotationCycleUpdateSchema = z.object({
  */
 
 export const loginSchema = z.object({
-  email: z.string().email(ERROR_MESSAGES.EMAIL_INVALID),
+  email: z.string().email(ERROR_MESSAGES.EMAIL_INVALID).trim(),
```
Comment (akiomatic) [inline src/constants/schema.ts]: The`.trim` should be before the `.email`.

File: src/constants/schema.ts [hunk 4]
```diff
@@ -158,7 +166,7 @@ export type TTenantInvitationSchema = z.infer<typeof tenantInvitationSchema>;
  * The schema for the shareHouse creation form
  */
 export const shareHouseCreationSchema = shareHouseNameSchema.extend({
-  startDate: z.string().datetime(),
+  startDate: z.string().datetime().trim(),
```
Comment (akiomatic) [inline src/constants/schema.ts]: The`.trim` should be before the `.datetime`.

**Review by akiomatic — CHANGES_REQUESTED** [review]
Thank you. Could you check my comments?

File: src/constants/schema.ts (hunk 4 above)
Comment (yuki-92c) [reply src/constants/schema.ts]: thank you for your review. I fixed according to the review so please check it again

**Review by akiomatic — APPROVED** [review]
Thank you. LGTM.

**Pull request merged**